src/services/review/RepoCheckerService.ts

```diff
@@ -53,17 +58,102 @@ export default class RepoCheckerService {
   constructor({
     siteMemberRepository,
     repoRepository,
-    sitesService,
     gitFileSystemService,
     git,
   }: RepoCheckerServiceProps) {
     this.siteMemberRepository = siteMemberRepository
-    this.sitesService = sitesService
+
     this.gitFileSystemService = gitFileSystemService
     this.git = git
     this.repoRepository = repoRepository
   }
 
+  isCurrentlyLocked(repo: string): ResultAsync<true, SiteCheckerError> {
+    const logsFilePath = path.join(SITE_CHECKER_REPO_PATH, repo)
+    // create logs folder if it does not exist
+
+    if (!fs.existsSync(logsFilePath)) {
+      fs.mkdirSync(logsFilePath, { recursive: true })
+    }
+
+    // check if checker.lock exists
+    const lockFilePath = path.join(logsFilePath, LOCK_CONTENT)
+
+    return ResultAsync.fromPromise(
+      fs.promises.access(lockFilePath, fs.constants.F_OK),
+      (err) => new SiteCheckerError(`${err}`)
+    ).map(() => true)
+  }
+
+  createLock(repo: string) {
+    const folderPath = path.join(SITE_CHECKER_REPO_PATH, repo)
+    if (!fs.existsSync(folderPath)) {
+      fs.mkdirSync(folderPath, { recursive: true })
+    }
+    // create a checker.lock file
+    const lockFilePath = path.join(folderPath, LOCK_CONTENT)
+    return ResultAsync.fromPromise(
+      fs.promises.writeFile(lockFilePath, "locked"),
+      (error) => new SiteCheckerError(`${error}`)
+    )
+  }
+
+  deleteLock(repo: string) {
+    const lockFilePath = path.join(SITE_CHECKER_REPO_PATH, repo, LOCK_CONTENT)
+    return ResultAsync.fromPromise(
+      fs.promises.unlink(lockFilePath),
+      (error) => new SiteCheckerError(`${error}`)
+    )
+  }
+
+  createErrorLog(repo: string, error: SiteCheckerError) {
+    const errorLogFilePath = path.join(
+      SITE_CHECKER_REPO_PATH,
+      repo,
+      REPO_ERROR_LOG
+    )
+    return ResultAsync.fromPromise(
+      fs.promises.writeFile(errorLogFilePath, error.message),
+      (err) => new SiteCheckerError(`${err}`)
+    ).mapErr((err) => {
+      logger.error(
+        `SiteCheckerError: Error creating error log for repo ${repo}: ${err}`
+      )
+      return err
+    })
+  }
+
+  isCurrentlyErrored(repo: string): ResultAsync<boolean, SiteCheckerError> {
+    const errorLogFilePath = path.join(
+      SITE_CHECKER_REPO_PATH,
+      repo,
+      REPO_ERROR_LOG
+    )
+    return ResultAsync.fromPromise(
+      fs.promises.access(errorLogFilePath, fs.constants.F_OK),
+      (err) => new SiteCheckerError(`${err}`)
+    )
+      .map(() => true)
```

Comment:
should this be okAsync(true)?

Reply:
are we confusing `map` with `andThen`? 

this would be similar to 
```
ResultAsync.fromPromise(
      fs.promises.access(errorLogFilePath, fs.constants.F_OK),
      (err) => new SiteCheckerError(`${err}`)
    )
      .andThen(() => okAsync(true))
      .orElse(() => okAsync(false))
```

preference for map here ya, less verbose